packages/workers-shared/asset-worker/src/utils/headers.ts

```diff
@@ -29,3 +32,48 @@ export function getHeaders(
 function isCacheable(request: Request) {
 	return !request.headers.has("Authorization") && !request.headers.has("Range");
 }
+
+export function attachCustomHeaders(
```

Comment:
Feels like this should be shared between pages and workers too?

Reply:
https://github.com/cloudflare/workers-sdk/pull/8279#discussion_r1985290531